src/main/kotlin/br/unipar/plano/domain/cobrancas/model/Cobranca.kt

```diff
@@ -0,0 +1,80 @@
+package br.unipar.plano.domain.cobrancas.model
+
+import br.unipar.plano.domain.cobrancas.valueobjects.StatusCobranca
+import java.math.BigDecimal
+import java.math.RoundingMode
+import java.time.LocalDate
+import javax.persistence.*
+
+
+@Entity
+class Cobranca(
+    @EmbeddedId
+    val id: IdCobranca,
+    @Column(nullable = false)
+    val valorContrato: BigDecimal,
+    @Column(nullable = false)
+    val valorAdicionalConsulta: BigDecimal,
+    @Column(nullable = false)
+    val valorAdicionalCirurgia: BigDecimal,
+    @Column(nullable = false)
+    val valorAdicionalIdade: BigDecimal,
+    @Column(nullable = false)
+    var status: StatusCobranca = StatusCobranca.ABERTO,
+    @Column(nullable = false)
+    val dataEmissao: LocalDate,
+    var dataCancelamento: LocalDate?,
+    @Column(nullable = false)
+    val dataVencimento: LocalDate,
+    @Column(nullable = false)
+    var valorTotal: BigDecimal?,
+    @ManyToOne(cascade = [CascadeType.ALL])
+    val contrato: Contrato
+) {
+    init {
+        valorTotal = valorContrato
+            .add(valorAdicionalIdade)
+            .add(
+                valorAdicionalCirurgia
+                    .multiply(BigDecimal.valueOf(contrato.cirurgias.count().toDouble()))
+            )
+            .add(
+                valorAdicionalConsulta
+                    .multiply(BigDecimal.valueOf(contrato.procedimentos.count().toDouble()))
+            )
+            .setScale(2, RoundingMode.HALF_UP)
+    }
+
+    fun cancelar(): Cobranca {
+        if (status != StatusCobranca.ABERTO && status != StatusCobranca.PAGO) {
```

Comment:
No mundo real a cobrança paga deve ser extornada e não cancelada.

```suggestion
        if (status != StatusCobranca.ABERTO) {
```